### ISSUE TITLE
Deleted Gmail label remains in N1

### ISSUE THREAD

**Issue by axllent** [nylas/N1#333]
After a label gets deleted in Gmail, the label (with original emails prior to label deletion) remains in N1.

**Comment by sethual** [comment]
Yeah have the same problem. Even when I deleted the account and relinked.

**Comment by mitjafelicijan** [comment]
+1

**Comment by gpleiss** [comment]
Same

**Comment by spang** [comment]
hey folks, thanks for the report! This is a known issue which we are working on resolving in the sync engine. (It's trickier than you'd think, because in Gmail IMAP, users can disable specific labels from appearing over IMAP, so we can't just use our regular folder list syncing code to determine that a label has been deleted, as it may have just been disabled in IMAP.)

**Comment by dkouvdis** [comment]
I have looked into the Gmail labels API, I understand what you (@spang) are saying. I dont have an advanced coding experience, I just want to point out a suggestion.

I am having the same issue, N1 is not reflecting the labels list I have over the API - I suggest creating a button somewhere in the preference, it should execute a specific job. Temporarily disable the account for ~30-60 seconds, request the labels list api, populating a fresh list of labels without modifying emails labels. 

It just that I dont understand, I removed all the cache in Application Support folder and reinstalled N1 to start fresh but it still persists. 

Further note: I guess the issue is coming from Nylas Sync Engine. Oh well
